jooqdb/src/main/java/org/cristalise/storage/jooqdb/JooqClusterStorage.java

```diff
@@ -1,339 +1,426 @@
+            config.setAutoCommit(false);
+            config.addDataSourceProperty( "autoCommit",false);
+            HikariDataSource newDs = new HikariDataSource(config);    
+            JooqHandler.ds = newDs;
+           
+        }
+    }
+
+    @Override
+    public void postStartServer() throws PersistencyException {
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postStartServer(DSL.using(nested));
+        });
+    }
+
+    @Override
+    public void postConnect() throws PersistencyException {
+        if(Gateway.getProperties().containsKey(JOOQ_AUTOCOMMIT)){
+            Gateway.getProperties().put(JOOQ_AUTOCOMMIT, true);
+        }
+        JooqHandler.connect().transaction(nested ->{
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.postConnect(DSL.using(nested));
+        });
+        
+    }
+
+    @Override
+    public void begin(Object locker) {
+        try {
+            DSLContext context = JooqHandler.connect();
+            contextMap.put(locker, context);
+
+            if (Logger.doLog(5)) JooqHandler.logConnectionCount("JooqClusterStorage.begin()", context);
+        }
+        catch (PersistencyException e) {
+            Logger.error(e);
+        }
+    }
+
+    @Override
+    public void commit(Object locker) throws PersistencyException {
+        DSLContext context = null;
+        
+        autoCommit =  Gateway.getProperties().getBoolean(JOOQ_AUTOCOMMIT);
+        if(!Objects.isNull(locker) && !contextMap.isEmpty()){
+            context = contextMap.get(locker) == null ? JooqHandler.connect() : (DSLContext) contextMap.get(locker);
+        } else {
+            context =  JooqHandler.connect();
+        }
+        context.transaction(nested -> {
+            for (JooqDomainHandler domainHandler : domainHandlers) domainHandler.commit(DSL.using(nested), locker);
+        });
+      
+        if (autoCommit) {
+            Logger.msg(1, "JooqClusterStorage.commit(DISABLED) - autoCommit:"+autoCommit);
+            return;
+        }
+
+        Logger.msg(1, "JooqClusterStorage.commit()");
+        try {
+            ((DataSourceConnectionProvider)context.configuration().connectionProvider()).acquire().commit();
+
+            if (Logger.doLog(5)) JooqHandler.logConnectionCount("JooqClusterStorage.commit()", context);
+        }
+        catch (Exception e) {
+            Logger.error(e);
+            throw new PersistencyException(e.getMessage());
+        }
+    }
+
+    @Override
+    public void abort(Object locker) throws PersistencyException {
+        DSLContext context = null;
+        
+        if(Objects.isNull(locker)){
+            throw new PersistencyException("Locker Object not found");
+        }
+        
+        if(contextMap.contains(locker)){
+            context = (DSLContext) contextMap.get(locker);
+        } else {
+            context  = JooqHandler.connect(); 

```

Comment:
What do we abort if we didn't find the context corresponding to the locker? We get a new connection from JooqHandler, but would there be any changes on that connection?

Reply:
Changes will be on next commit. Added a checker if the autocommit is set to false and the locker is not present the map, then will throw an error message